addon/components/dfp-ad.js

```diff
@@ -136,15 +141,29 @@ export default Component.extend({
         ad.defineSizeMapping(sizeMapping.build());
         this.set('mqList', mqList);
       }
+
+
       ad.addService(googletag.pubads());
+
       googletag.pubads().addEventListener('slotRenderEnded', (event) => {
         if (ad === event.slot) {
           run(() => {
-            if (event.isEmpty && this.get('clearOnEmptyRefresh')) {
-              googletag.pubads().clear([ad]);
+            let adState = {};
+            if (event.isEmpty) {
+              if (this.get('clearOnEmptyRefresh')) {
+                adState.isEmpty = true;
+                googletag.pubads().clear([ad]);
+              } else {
+                adState.isEmpty = this.adState.isEmpty;
+              }
+            } else {
+              adState.isEmpty = true;
```

Comment:
shouldn't this be `adState.isEmpty = false;`?

also, if I'm reading this, correctly, couldn't you do this at the top of this block? the logic reads a bit convoluted otherwise.
```js
let adState = {isEmpty: event.isEmpty};
```

Reply:
If clearOnEmptyRefresh isn't set, an empty response means the ad doesn't update. In that case we don't want to change the status at all. I'm going to reorganize this section though to make this easier to follow though.